### PR TITLE
Fix Move Identity creation warning

### PR DESCRIPTION
**Pull request by UMR1352**
# Description of change
Fix _Move_ lint warning that appeared when creating an Identity.

### CODE REVIEW

File: identity_iota_core/packages/iota_identity/sources/identity.move [hunk 1]
```diff
@@ -63,12 +63,12 @@ module iota_identity::identity {
         version: u64,
     }
 
-    /// Creates a new DID Document with a single controller.
+    /// Creates a [`Identity`] with a single controller.
```
Comment (wulfraem) [inline identity_iota_core/packages/iota_identity/sources/identity.move:66]:
```suggestion
    /// Creates an [`Identity`] with a single controller.
```

File: identity_iota_core/packages/iota_identity/sources/identity.move [hunk 2]
```diff
@@ -95,15 +103,19 @@ module iota_identity::identity {
...
             version: PACKAGE_VERSION,
-        }
+        };
+        let id = object::id(&identity);
+        transfer::share_object(identity);
+
+        id
     }
 
     /// Creates a new DID Document controlled by multiple controllers.
```
Comment (wulfraem) [inline identity_iota_core/packages/iota_identity/sources/identity.move:121]:
```suggestion
    /// Creates an [`Identity`] controlled by multiple controllers.
```

File: identity_iota_core/packages/iota_identity/sources/identity.move [hunk 3]
```diff
@@ -79,12 +79,20 @@ module iota_identity::identity {
...
+            version: PACKAGE_VERSION,
+        };
+        let id = object::id(&identity);
+        transfer::share_object(identity);
 
-        identity
+        id
     }
 
     /// Creates a new `Identity` wrapping DID DOC `doc` and controller by
```
Comment (wulfraem) [inline identity_iota_core/packages/iota_identity/sources/identity.move:98]:
```suggestion
    /// Creates a new [`Identity`] wrapping DID DOC `doc` and controller by
```

**Review by wulfraem — CHANGES_REQUESTED** [review]
Nitpicking ftw. ^^

Just suggesting aligning the `new.*` functions. ^^

File: identity_iota_core/packages/iota_identity/sources/identity.move (hunk 3 above)
Comment (UMR1352) [reply identity_iota_core/packages/iota_identity/sources/identity.move:98]: Mmm it seems _Move_ inline docs do not work the same as Rust's, using `[<ref to a piece of code>]` doesn't create a link.

**Review by wulfraem — APPROVED** [review]

File: identity_iota_core/packages/iota_identity/sources/identity.move (hunk 3 above)
Comment (wulfraem) [reply identity_iota_core/packages/iota_identity/sources/identity.move:98]: Ah, okay, just rolled with the updated version from this PR. But good to know. ^^

**Pull request merged**